src/TokenCache.ts

```diff
@@ -78,14 +83,17 @@ class TokenCache {
         };
 
         return getAccessToken(config)
-          .then((token) => getTokenInfo(this.oauthConfig.tokenInfoEndpoint, token.access_token))
           .then((token) => {
 
-            token.local_expiry = Date.now() + token.expires_in * 1000 - EXPIRE_THRESHOLD;
-            this._tokens[tokenName] = token;
+            const localExpiry = Date.now() + (token.expires_in * 1000 * DEFAULT_PERCENTAGE_LEFT);
```

Comment:
should be `const localExpiry = Date.now() + (token.expires_in * 1000 * (1 - DEFAULT_PERCENTAGE_LEFT));`